src/hdf5/spire_2ioda.py

```diff
@@ -0,0 +1,281 @@
+    for n, iodavar in enumerate(obsvars):
+        # set the obs space attributes
+        varDict[iodavar]['valKey'] = iodavar, obsValName
+        varDict[iodavar]['errKey'] = iodavar, obsErrName
+        varDict[iodavar]['qcKey'] = iodavar, qcName
+        varAttrs[iodavar, obsValName]['coordinates'] = 'longitude latitude'
+        varAttrs[iodavar, obsErrName]['coordinates'] = 'longitude latitude'
+        varAttrs[iodavar, qcName]['coordinates'] = 'longitude latitude'
+        varAttrs[iodavar, obsValName]['units'] = obsvars_units[n]
+        varAttrs[iodavar, obsErrName]['units'] = obsvars_units[n]
+
+    # Set units of the MetaData variables and all _FillValues.
+    for key in meta_keys:
+        dtypestr = MetaDataKeyList[meta_keys.index(key)][1]
+        if MetaDataKeyList[meta_keys.index(key)][2]:
+            varAttrs[(key, metaDataName)]['units'] = MetaDataKeyList[meta_keys.index(key)][2]
+        varAttrs[(key, metaDataName)]['_FillValue'] = missing_vals[dtypestr]
+        ioda_data[(key, metaDataName)] = np.array(obs_data[key], dtype=dtypes[dtypestr])
+
+    # Transfer from the 1-D data vectors and ensure output data (ioda_data) types using numpy.
+    for n, iodavar in enumerate(obsvars):
+        ioda_data[(iodavar, obsValName)] = np.array(obs_data[iodavar], dtype=np.float32)
+#        ioda_data[(iodavar, obsErrName)] = np.full(ntotal, obserrlist[n], dtype=np.float32)
+        ioda_data[(iodavar, qcName)] = np.full(ntotal, 2, dtype=np.int32)
+
+    # setup the IODA writer
+    writer = iconv.IodaWriter(args.output_file, MetaDataKeyList, DimDict)
+    # write everything out
+    writer.BuildIoda(ioda_data, VarDims, varAttrs, GlobalAttrs)
+
+
+def get_data_from_file(afile, col_names, file_name):
+
+    # Get instrument reference
+    instrument_ref = afile.attrs['tx_id'].decode('UTF-8') 
+
+    # Pull each data type (variable) and create a list
+    latitude = [afile['sp_lat'][ii] for ii in range(len(afile['sp_lat']))]
+    longitude = [afile['sp_lon'][ii] for ii in range(len(afile['sp_lon']))]
+    dateTime = [int(afile['sample_time'][ii]) for ii in range(len(afile['sample_time']))]  # datetime with different ref time
+    windSpeedAt10M = [afile['wind'][ii] for ii in range(len(afile['wind']))]
+    sensorIdentification = [instrument_ref]*len(latitude)
+
+    # Make a column to have a constant elevation for the "station"
+    height = [10]*len(dateTime)
+
+    # Make a list of lists to feed into dataframe
+    data_lists = list(zip(latitude, longitude, dateTime, sensorIdentification,
+                          height, windSpeedAt10M))
+
+    # All observation data for this file to append to the master dataframe
+    obs_data_append = pd.DataFrame(data_lists, columns=col_names)
+    return obs_data_append
+
+
+def adjust_dateTime(obs_DF, dat_ref):
+    obs_DF['dateTime'] = obs_DF['dateTime']+int(dat_ref)
+    return obs_DF
+
+def adjust_longitude(obs_DF):
+    obs_DF['longitude'] = (obs_DF['longitude']+int(180)) % 360 - int(180)
```

Comment:
```suggestion
    obs_DF['longitude'] = (int(obs_DF['longitude'])+180) % 360 - 180
```

Reply:
I full agree with Pat on unimportant usage of `int` casting.  When not needed, simply don't cast.

Reply:
Thanks @PatNichols. Not sure on c++ behavior - is `180` automatically an `int` since there's no decimal (i.e. `180.` would behave as a float). I don't think we want to cast the longitude into an `int` though. The decimal place in latitude in needed to be retained.

Reply:
so maybe this is fine?
```obs_DF['longitude'] = (obs_DF['longitude']+180) % 360 - 180```

Reply:
@smaticka if you don't cast obs_DF to an int then the mod operator will be wrong. Probably what you want is math.fmod(obs_DF['longitude']+180,360) ?

Reply:
oh nice, yeah I think that's what I need. I need longitude to stay a float (then add exactly 180 to it, then take the modulus with 360, then subtract 180 from the result of the modulus. So would it be this?:
`obs_DF['longitude'] = math.fmod(obs_DF['longitude']+180,360) - 180`

Reply:
weirdly it worked before.. when I added int(180) to the float.. I checked the plots. The idea was to shift the data into the proper range (from 0, 360 to -180, 180):
![Untitled](https://prod-files-secure.s3.us-west-2.amazonaws.com/59d02984-8f43-4e66-bf9d-1039571a6cc7/400b6108-d922-4141-a88d-ce01e86bd089/Untitled.png)

Reply:
Why not  (value - 180) ? Assuming value is between 0 to 360 -> -180 to 180

Reply:
b/c 0 to 180 retain their values. but the range 181 to 360 becomes -180 to -1 (359 becomes -1 for example)

Reply:
`math.fmod()` doesn't work on pandas series, so I used the pandas series method. (though I don't understand why the original method wasn't ok).

Reply:
@smaticka hmmm I get it now... Too little sleep.
What you want then is 
      if ( long > 180):
          long = -360 + long

Reply:
@PatNichols 
An `if` statement would require a `for` loop. `mod`, which is a `pandas` `method`, is presumably vectorized, which should be more efficient than a for loop. Is there a problem with me using the `mod` method?